Homework 4

TelefunkenU87 opened the pull request:
I added the code for data validation within the program. I think it is all done but I just have one problem that I thought would be better to work through in a pull request instead of trying to sort it out in the discord.

When I added the functions to checks.cs I was unable to compile. The reason that the compiler gave me was that because of the new validation, "not all code paths return a value"

What should I do to resolve this?

Almantask commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4907993&bid=16250230)

Almantask commented on Src/BootCamp.Chapter/Lesson4.cs:
```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4907993)

Almantask commented on Src/BootCamp.Chapter/Lesson4.cs:
```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4907993)

Almantask commented:
Tests don't pass and I don't care- bad spec- you are allowed to make mistakes. Freedom :)

Almantask commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4907993&bid=16250230)

TelefunkenU87 commented:
Ok, I removed the unused 'using' lines in program.cs and Lesson4.cs. I then added the functions to Checks.cs anyways just to see how codacy and azure pipeline would interpret them. I took Almantask's advice from the lesson3 PR and removed all the unnecessary white space lines in the code for this lesson 4 RP.

Almantask commented on Src/BootCamp.Chapter/Lesson4.cs:
```diff
@@ -0,0 +1,95 @@
+            }
+        }
+
+        public static void NewPerson()
```
GetPersonData or similar should be a function name. Method names = verbs, stick to it.
